app/controllers/unsubscriptions_controller.rb

```diff
@@ -28,13 +36,31 @@ def confirmed
 
   def set_attributes
     @id = params.require(:id)
-    @subscription = api.get_subscription(@id)
+    @original_subscription = api.get_subscription(@id)
+    @latest_subscription = api.get_latest_matching_subscription(@id)
+    @subscription = @latest_subscription
     @title = @subscription.dig("subscription", "subscriber_list", "title").presence
     @authenticated_for_subscription = check_authenticated(@subscription)
   end
 
-  def subscription_ended?
-    @subscription.dig("subscription", "ended_at").present?
+  def subscription_is_current
+    @latest_subscription.id == @original_subscription.id
+  end
+
+  def old_subscription_matches_current
+    @latest_subscription.frequency == @original_subscription.frequency
+  end
+
+  def subscription_ended?(subscription)
+    subscription.dig("subscription", "ended_at").present?
+  end
+
+  def already_unsubscribed
+    subscription_ended(@latest_subscription)
+  end
+
+  def unsubscribe_current
+    # do nothing - yield to the confirm.html render
```

Comment:
This is a bit of a code smell. I really wanted this method to exist because it makes the `confirm` method logic really easy to read. My understanding is that Rails will yield to whatever view matches the current controller's public method name routed via `routes.rb`, provided you don't `return` anything in that method.